src/test/scala/is/hail/expr/ir/PruneSuite.scala

```diff
@@ -316,6 +316,18 @@ class PruneSuite extends SparkSuite {
       Array(subsetMatrixTable(mat.typ, "global.g1", "va.r3", "sa.c3"), null))
   }
 
+  @Test def testMatrixAnnotateRowsTableMemo() {
+    val tl = TableLiteral(MatrixRowsTable(mat).execute(hc))
+    val mart = MatrixAnnotateRowsTable(mat, tl, "foo", None)
+    checkMemo(mart, subsetMatrixTable(mart.typ,"va.foo.r3", "va.r3"),
+      Array(subsetMatrixTable(mat.typ, "va.r3"), subsetTable(tl.typ, "row.r3")))
+
+    val mart2 = MatrixAnnotateRowsTable(mat, tl, "foo",
+      Some(FastIndexedSeq(GetField(GetField(Ref("va", mat.typ.rvRowType), "r2"), "x"))))
+    checkMemo(mart2, subsetMatrixTable(mart2.typ,"va.foo.r3", "va.r3"),
+      Array(subsetMatrixTable(mat.typ, "va.r3", "va.r2.x"), subsetTable(tl.typ, "row.r3"), null))
```

Comment:
why include the trailing `null`? It seems we could just list the two children we care about.

Reply:
bump

Reply:
ah, sorry. The array is expected to be the same length as `Children`, with nulls used to designate IRs we don't want to check for type equality.

We could redesign this to use a `zip` with shorter arrays instead of nulls.